packages/@aws-cdk/aws-appsync/lib/schema-intermediate.ts

```diff
@@ -296,3 +296,102 @@ export class InputType implements IIntermediateType {
     this.definition[options.fieldName] = options.field;
   }
 }
+
+/**
+ * Properties for configuring an Union Type
+ *
+ * @props definition - the object types for this union type
+ *
+ * @experimental
+ */
+export interface UnionTypeProps {
```

Comment:
naming convention: the `props` suffix is used for constructs. we use `options` as the suffix otherwise.